redash/utils/json.py

```diff
@@ -0,0 +1,71 @@
+import binascii
+import decimal
+import json
+import uuid
+from datetime import date, datetime, time, timedelta
+
+from sqlalchemy.orm.query import Query
+
+from redash.query_runner import query_runners
+
+
+class JSONEncoder(json.JSONEncoder):
+    """Adapter for `json.dumps`."""
+
+    def __init__(self, **kwargs):
+        self.encoders = [
+            r.custom_json_encoder for r in query_runners.values() if hasattr(r, "custom_json_encoder") and r.enabled
```

Comment:
I notice this line seems to be the only actual change in this PR. @AndrewChubatiuk , please undo all the other changes - e.g. creating this json.py file and all the import changes, as well as the classmethod additions. Work like this doesn't make it easy to review code, nor does it make it easy to follow code when investigating issues.

Reply:
Additionally, please explain this change in the PR description in two or three paragraphs, providing:
- context of why this change is necessary
- what led to this change (including links to all previous related PRs)
- what the current behavior is, and why it's wrong
- and what the intended behavior is.
 
Additionally, please describe how you've tested it. If you only have manual tests, then please add some kind of automated test with pytest or unittest to make sure that this does not break again.

Reply:
i cannot make this change without moving JSONEncoder to a separate file. It will lead to circular import problem as `query_runners` variable is inside `redash`, which uses `redash.utils` and JSONEncoder is inside `redash.utils` module, which uses `redash module`

Reply:
@guidopetri also updated PR description

Reply:
Can you try making the import of `query_runners` that you need within the JSONEncoder definition? I believe that will circumvent the circular import.

Reply:
@guidopetri commited in a way you want. you can check CI logs

Reply:
I understand that the current import location leads to a circular import. Please read my comment above more carefully - can we make the import within the JSONEncoder definition?

Reply:
done

Reply:
@guidopetri tried but imports, which cause circular dependencies issues inside classes, but it didn't work. moved `query_runners` to another module to have less changes comparing to moving json methods to a separate module